### ISSUE TITLE
[FR] Option for limited history lifetime

### ISSUE THREAD

**Issue by NicoHood** [dino/dino#67]
It would be cool to have a feature that limits the lifetime of your history. E.g. 7 days or so. Also I am wondering if the messages are encrypted on my disk and how the key is saved.

**Comment by mar-v-in** [comment]
Dino does not encrypt messages stored on your disk. If you want to have your messages encrypted, you should use disk encryption (luks) or directory encryption (ecryptfs). The history, account passwords and your omemo keys are stored in the directory `$HOME/.local/share/dino`.

If you want to encrypt messages to prohibit other apps on your device to access them (e.g. your browser) you can use tools like [firejail](https://firejail.wordpress.com/) to cleanly separate possible vulnerable apps. This seems also to be partly a reason for your request #66, so you might want to consider this anyway.

The request for history lifetime limitation is still valid though ;)

**Comment by NicoHood** [comment]
Even though alternatives like firejail are available I think it is still a good idea to keep messages secure. I already have full disk encryption, but encrypted messages would still be a feature I wish. And especially a limited history could solve this partly (for now).

#66 is also a general issue. You cannot and dont want to put everything inside a firejail sandbox. Also just because other might want to login on your pc. And password remembering should not be enabled by default.

**Comment by mejo-** [comment]
I agree that an option for limited history lifetime - and even more important - an option to completely turn of local history for specific contacts/channel is a crucial option for users who care about privacy.

**Comment by martin21** [comment]
Being able to delete history is an important privacy feature. A chat buddy requested to me to delete chat history from time to time, yet, currently, I can't. I am trying to figure out the SQLite database. I find the messages with `select * from message;`, however I see no key that is unique for the chat partner. I bet I need to build a relation with another table.

Can you provide any hint how to manually delete the chat history of one chat with SQLite3 but not the chat itself – except maybe the last few days? I understand you cannot implement everything at once, however I just got that request of the chat buddy again and for me it is a completely reasonable request.

**Comment by martin21** [comment]
Found it, appears to work like `SELECT * FROM message WHERE counterpart_id = SOMEID;`

**Comment by MainTomato** [comment]
I agree that being able to disable local history would be preferred. Not only for privacy but mental well-being as well, I would even argue that it should be the default.

**Comment by mbeko** [comment]
The feature is useful if someone wants to limit the space used by an app or if they want to avoid storing illegal material on the given device in the long term. 

But privacy can hardly be an argument considering that the messages can be floating around on your and other people's devices. I'm also not sure how it helps mental well-being when you don't know who else still has access to the messages and what they'll do with it.

**Comment by datainstigator** [comment]
Can I input something here please:

When I started using DinoIM, I was struck by the lack of 'Delete History' functionality - as were my friends (I counted 14 friends in the group, all from different demographics, etc, all who like to keep in touch on our Linux laptops). It was the primary reasons we ditched DinoIM, a couple of years back, and went with something else. I see now still, that it hasn't been implemented.

Why would this be a 'we need to justify why we need to delete history'? 

I'm a senior software developer, if I was working on a privacy focused messaging app like Dina, the question would be: 
> 'why have we NOT got 'delete history' functionality??? Next sprint: get it done, and get it done yesterday'.

Even the 'high street' messaging apps like Signal (etc) have implemented this functionality .
Not only should we not be asking  'why?', surely we should be asking 'why not?'. 

Please introduce this, this sort of change request is not difficult to implement.

Thanks :)

**Comment by licaon-kter** [comment]
@datainstigator
> this sort of change request is not difficult to implement.
which PR is yours? :+1:

**Comment by datainstigator** [comment]
Haha.

Yes but you avoided the question :)

Why not implement this?

**Comment by mbeko** [comment]
If you're a senior software developer and you're adding features without determining the need methodically, evaluating the actual effort, considering the usability or prioritising in relation to other issues, that rather diminishes the weight of your statements instead of increasing it.

Please explain how deleting history provides privacy in the context of the internet and devices you have no control over. I can save, copy and distribute your messages before you delete them as much as I please and you would be wrongly thinking they're gone.

The feature can be added to reduce the risk and make it more convenient, but that doesn't make it a high priority or privacy-relevant issue.

You also seem to forget that this is not a company but an open source project maintained by two developers in their free time. As a senior software developer, you should know that open source works by making contributions in form of code or informed arguments, not rants based on your personal preferences.

**Comment by licaon-kter** [comment]
@mbeko buy an ENTER key c'mon :)

**Comment by datainstigator** [comment]
Hi @mbeko hope you're well.

Inline reply:

If you're a senior software developer and you're adding features without determining the need methodically, evaluating the actual effort, considering the usability or prioritising in relation to other issues, that rather diminishes the weight of your statements instead of increasing it.

> I think determining the need methodically is pretty simple, it goes approximately like this; (depending on which MAM system you're using and various other environmental factors which I may not be aware of in your case), I haven't looked at the Dino source:

> Do users want the feature? In this case, yes, there are outstanding requests dating back more than 2 years.

> Are their other aspects of the codebase that make this difficult to implement, or do we have more pressing issues? Well maybe - that's a simple answer you can give, no need for antyhing but a one-liner, which might be: "It's something we've looked at, but it's down on our list ATM".

...

> From your message, am I correct in evaluating that you have more pressing feature requests? It's hard to tell from your wording. If that's the case, that's all you need say, I get it, we're all busy. But if we can't post questions here, where can we post them? :)


Please explain how deleting history provides privacy in the context of the internet and devices you have no control over. (1)

> Sure :) In this situation I think the logical thing to do is take a wider, calmer view; focus on the devices that we DO have control over, and make sure things are as user-oriented, and as flexible as possible. In this case, empowering a user to be able to delete content from their device seem (at least on surface) to be entirely reasonable. I also think (and I tell myself this as much as possible on every project I work on), that we should look at what other teams are doing on similar projects, and try to emulate the best features. Do you disagree with my reasoning, or are you just too busy or concerned with more important issues?


Please explain how deleting history provides privacy in the context of the internet and devices you have no control over. (2)

> Dino encrypts messages that go out over the wire, right? So we're not concerned with people intercepting them on the internet, as they can't read them. So I don't think the internet is relevant to the feature request. Do you disagree with this logic? If so fine, I'm always willing to admit I'm wrong :)


I can save, copy and distribute your messages before you delete them as much as I please and you would be wrongly thinking they're gone.

> If those messages are properly encrypted you can copy and distribute them as much as you like, but you won't know what they said unless you have access to my local device - and if I have the ability to delete the messages from my device, I rightly know that they're gone from my device. No?


The feature can be added to reduce the risk and make it more convenient, but that doesn't make it a high priority or privacy-relevant issue.

> Ah, so you are agreeing that it does indeed reduce risk (and therefore enhance privacy), by adding the 'delete history' functionality? If we agree, then that's great, I'm not being facetious. So building a messaging app that makes privacy a high priority, why would this not be a priority?


You also seem to forget that this is not a company but an open source project maintained by two developers in their free time. 

> Hmm, I don't think I've forgotten anything, I'm just posting what I considered 'food for thought', I didn't mean to elicit a defensive response. My apologies if I've annoyed you. I'm just trying to pose a position which I (and a lot of people it seems, since apps like Signal who pride themselves on listening to their userbase, have implemented vanishing messages) think is desirable.


As a senior software developer, you should know that open source works by making contributions in form of code or informed arguments, not rants based on your personal preferences.

> I do know how it works, but thanks for reiterating it for me.

> Rants? Well, hmm... I don't think I was ranting, but my apologies if you thought I was. Can we get back to the point in hand? The question is: What's the reason for not implementing 'delete history'? If you have a reason, fine! Let's further the discussion at least?


...

You could be dealing with chunks of legacy code that lack flexibility, been there and done that!

I don't' know which MAM framework you're using. But: <Retrieval, storage, UI/alert output, LocalDelete -> Sync, etc.> is a (very) high level overview of the process, I think.

But anyway look, the how-to's are irrelevant. If it's not high on your list then that's all that's needed to say. Be interesting to know why it isn't though...

**Comment by mbeko** [comment]
> I think determining the need methodically is pretty simple, it goes approximately like this

There are proven standards for requirement elicitation. No offense, but I'm not interested in discussing the topic with someone who's not familiar with them.

> But if we can't post questions here, where can we post them? :)

For one, [a more appropriate issue would be a good start](https://github.com/dino/dino/issues/472).

The problem was not that much with the question itself, but how you presented it in a demanding way focussing only on yourself and applying your standards.

> So we're not concerned with people intercepting them on the internet, as they can't read them

If you send a message to one or multiple people, they can read them. Because of the internet, you don't know where in the world on how many devices the messages arrive and who actually reads and/or copies them. When you delete them on your device, you don't know where else they float around, so you don't gain much privacy-wise.

> I don't think I've forgotten anything, I'm just posting what I considered 'food for thought'

Your comment about implementing the feature yesterday was hardly food for thought. It rather came off as if you wanted to pressure people who work for free into fulfilling your wishes as soon as possible. Likewise just claiming that it's easy to implement instead of actually doing it is usually perceived as rude.

> Signal who pride themselves on listening to their userbase, have implemented vanishing messages

You can't compare a centralised service like Signal with a diverse, federated system like XMPP, especially not when it comes to privacy-relevant deletion. Both clients and servers on the others side might not support deletion requests.

> If it's not high on your list then that's all that's needed to say

That's exactly what I said except for the fact that it's not my list. You even quoted it.

**Comment by unDocUMeantIt** [comment]
it is possible to be the member of a group chat that gets visited by jerks who leave material that might get you into legal trouble if it is found on your device. in such a case, it ist utterly **important** to be able to remove this stuff entirely from your devices.

**Comment by OperationNightWalker** [comment]
@datainstigator - your messages on this topic were absolutely spot on, sorry for the childish replies you got here from @mbeko, that's pretty embarrassing for him tbh.

@mbeko we absolutely need this basic functionality in a messaging app that pretends to address privacy concerns.  

@unDocUMeantIt EXACTLY!! I Think we have a project run by juniors, my advice: jump ship! There are so many better options out there in 2026, this project should just die not tbh they had their chance...
